### DISCUSSION TITLE
Partial ticket cancellation

### DISCUSSION THREAD

**Discussion by chpatrick** [pretix/pretix#4695] [Ideas]
Hi, is there any plan to support cancelling only some of the tickets you've bought instead of the whole order? I.e. you bought five but two people can't make it so you cancel their two tickets. This is a somewhat common occurance in our case and leads to people just not cancelling, which wastes seats. To simplify things it would be fine just for free events without refunds etc.

**Comment by raphaelm** [comment]
We'd love to have it, but it's currently not on our plan for the near future as it is quite complex to implement. The complexity is less in the refunds, but more in the validation of when a partial cancellation is possible and what happens e.g. in case a group discount is used etc

**Comment by AlexanderWillner** [comment]
Technically, implementing partial ticket cancellation in the frontend might be straightforward, particularly for free events. In our specific case:

- Free Events: Since our events are free of charge, there are no complications related to refunds, discounts, or payment adjustments.
- Backend Functionality: Partial ticket cancellation is already supported via the backend by editing a ticket directly through the following link: https://pretix.eu/control/event/XXX/YYY/orders/ZZZ/change
- Current Limitation: End users do not have access to this functionality in the frontend. As a result, “customers” are required to email us to request partial ticket cancellations.

**Reply by raphaelm** [reply]
I'm sure it is straight-forward technically when restricting to the case of only the simplest configurations of pretix and not supporting all other cases, sure.

**Comment by AlexanderWillner** [comment]
I’ve put together a WIP/RFC patch that adds partial cancellation support:
https://github.com/AlexanderWillner/pretix/commit/565cc1a1fec48551a37409e104f138b8afbbe6bf

It runs fine on my setup and already includes German translations. I’m not deeply familiar with the pretix contribution workflow, so feedback on direction or required adjustments would be useful. At this stage it’s probably practical for free orders only.

<img width="2860" height="1628" alt="pretix-partial-cancellation" src="https://github.com/user-attachments/assets/30cace57-5220-458b-b1bf-2ca4cad3cdfa" />

**Comment by AlexanderWillner** [comment]
As per #5672, I created a plugin at https://marketplace.pretix.eu/products/partial-cancellation, however, TIL that plugins not developed by Pretix can't be used in the hosted version.
As #5672 will not be merged in its current form, I'd like to discuss with @raphaelm (and/or probably others) how this feature should look like so we can get this merged. For me it's fine to have it as an optional, double guarded feature for free orders only to get started ;)
The current feedback how I see it:

* [ ] we've to figure out how how to make it "look good" (current status above)
* [ ] the plugin needs to be integrated into the OrderPositionChange view instead of being its own flow / form
  * From a user's perspective it is weird if "cancel an addon" is in a different place than "cancel a ticket". We need one unified view that allows making changes to the order, not a number of them where the ticket buyer needs to understand specific verbiage to know which one is which.
* [ ] we need to deal with restrictions like min_per_order
* [ ] a safe initial plan could be to validate those restrictions strictly, i.e. do not allow partial cancellations that would lead to Item.min_per_order be violated. For example:
  * consider a product "Group Ticket" that has min_per_order = 20 set so one can only get 20 or more tickets. This becomes pretty pointless if, after the purchase is complete, one can just remove 10 tickets again and end up with an order of 10 tickets. 
  * same for Voucher.min_usages, discounts (that one is less relevant for free orders), etc. 
  * Option: Never allow partial cancellations if one of these features is used? Probably not. Completely ignore these settings? Probably not. Dozens of new settings? Probably not. I do not know the answer to this question, and we tend not to build features into core that we do not yet know how they interact with the rest of the system "properly".
* [ ] ideally, extend this to paid orders in a later version

**Reply by raphaelm** [reply]
Just FYI @bockstaller is working on this topic from a different by refactoring how pretix decides what tickets are cancellable and at what cancellation fee. In the end, this will probably also lead us to building something for this

**Reply by AlexanderWillner** [reply]
> Just FYI @bockstaller is working on this topic from a different by refactoring how pretix decides what tickets are cancellable and at what cancellation fee. In the end, this will probably also lead us to building something for this

Hey @bockstaller, nice to hear that! Is there already something I could test? And are any of my changes useful for this?

**Comment by AlexanderWillner** [comment]
Thanks for the update, so I will pause working on this for now.

**Comment by AlexanderWillner** [comment]
I saw https://github.com/pretix/pretix-withdrawal to comply with EU legislation. But I couldn't find any information about a partial cancellation / partial withdrawal. Is there anything to test already?
